packages/amazonq/src/lsp/client.ts

```diff
@@ -160,120 +160,141 @@ export async function startLanguageServer(
 
     const disposable = client.start()
     toDispose.push(disposable)
+    await client.onReady()
 
-    const auth = new AmazonQLspAuth(client)
+    const auth = await initializeAuth(client)
 
-    return client.onReady().then(async () => {
-        await auth.refreshConnection()
+    await postStartLanguageServer(auth, client, resourcePaths, toDispose)
 
-        if (Experiments.instance.get('amazonqLSPInline', false)) {
-            const inlineManager = new InlineCompletionManager(client)
-            inlineManager.registerInlineCompletion()
-            toDispose.push(
-                inlineManager,
-                Commands.register({ id: 'aws.amazonq.invokeInlineCompletion', autoconnect: true }, async () => {
-                    await vscode.commands.executeCommand('editor.action.inlineSuggest.trigger')
-                }),
-                vscode.workspace.onDidCloseTextDocument(async () => {
-                    await vscode.commands.executeCommand('aws.amazonq.rejectCodeSuggestion')
-                })
-            )
-        }
+    return client
+}
 
-        if (Experiments.instance.get('amazonqChatLSP', true)) {
-            await activate(client, encryptionKey, resourcePaths.ui)
-        }
+async function initializeAuth(client: LanguageClient): Promise<AmazonQLspAuth> {
+    AuthUtil.instance.regionProfileManager.onDidChangeRegionProfile(async () => {
+        void pushConfigUpdate(client, {
+            type: 'profile',
+            profileArn: AuthUtil.instance.regionProfileManager.activeRegionProfile?.arn,
+        })
+    })
 
-        const refreshInterval = auth.startTokenRefreshInterval(10 * oneSecond)
+    const auth = new AmazonQLspAuth(client)
+    await auth.refreshConnection(true)
+    return auth
+}
 
-        const sendProfileToLsp = async () => {
-            try {
-                const result = await client.sendRequest(updateConfigurationRequestType.method, {
-                    section: 'aws.q',
-                    settings: {
-                        profileArn: AuthUtil.instance.regionProfileManager.activeRegionProfile?.arn,
-                    },
-                })
-                client.info(
-                    `Client: Updated Amazon Q Profile ${AuthUtil.instance.regionProfileManager.activeRegionProfile?.arn} to Amazon Q LSP`,
-                    result
-                )
-            } catch (err) {
-                client.error('Error when setting Q Developer Profile to Amazon Q LSP', err)
-            }
+async function postStartLanguageServer(
```

Comment:
nit: should we call this something like `OnLanguageServerReady` to be consistent with Flare language?